data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/WorkerCommonUtils.java

```diff
@@ -14,14 +14,26 @@
 import org.opensearch.dataprepper.plugins.source.opensearch.OpenSearchSourceConfiguration;
 
 import java.time.Duration;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
+import static com.google.common.math.LongMath.pow;
+import static com.google.common.primitives.Longs.min;
+
 public class WorkerCommonUtils {
+    private static final Random RANDOM = new Random();
+
+    static final Duration BACKOFF_ON_EXCEPTION = Duration.ofSeconds(60);
 
     static final int ACKNOWLEDGEMENT_SET_TIMEOUT_SECONDS = Integer.MAX_VALUE;
+    static final Duration STARTING_BACKOFF = Duration.ofMillis(2500);
```

Comment:
Is there any reason not to make this sub-second?

Reply:
I don't think there's any reason not too. I just figured in most cases the supplier would still be going but I could start it at 0.5 seconds?